ui/app/serializers/transform/role.js

```diff
@@ -1,10 +1,12 @@
 import ApplicationSerializer from '../application';
 
 export default ApplicationSerializer.extend({
+  primaryKey: 'name',
   extractLazyPaginatedData(payload) {
     return payload.data.keys.map((key) => {
       const model = {
         id: key,
+        name: key,
```

Comment:
I noticed we kept `id: key` here when adding `name` - is that something we should also do in `ui/app/serializers/role.js` on line 17?

Reply:
That's a good question, I went back and forth between keeping both and replacing it. I reasoned to add to this one because `extractLazyPaginatedData` seems to be vault-specific, so there's no guarantee that we're using it in a conventional way which would take advantage of `primaryKey` (which I believe is extracted via [extractId](https://api.emberjs.com/ember-data/4.9/classes/JSONAPISerializer/methods/extractId?anchor=extractId))
I also figured by removing it from the other serializers where we use `primaryKey` there may be fewer places we need to track down in case we need to update the ID in the future. But, I'm happy to add back `id` to the others if the team thinks it makes the most sense!

Reply:
Thanks for explaining your reasoning! I think this makes sense, and ideally decreases the difficulty in tracking down future ember data issues